### PR TITLE
Provide orphan `ToJSON` instances for `HardForkNodeToClientVersion` and `HardForkNodeToNodeVersion`

### PR DESCRIPTION
**Pull request by dnadales**
Required for https://github.com/input-output-hk/cardano-node/issues/5000

### CODE REVIEW

**Comment by dnadales** [comment]
Blocked on https://github.com/input-output-hk/cardano-node/pull/5013 (we should wait till this one is merged).

**Comment by dnadales** [comment]
@coot done.

@Jimbo4350 let me know if this can be merged.

File: cardano-node/test/cardano-node-test.hs [hunk 1]
```diff
@@ -12,20 +12,25 @@ import qualified Test.Cardano.Node.FilePermissions
...
+      , Test.Cardano.Node.POM.tests
+      , Test.Cardano.Node.FilePermissions.tests
+      ]
 #else
-    [ Test.Cardano.Node.Json.tests
-    , Test.Cardano.Node.POM.tests
-    ]
+      [ Test.Cardano.Node.Json.tests
+      , Test.Cardano.Node.POM.tests
+      ]
```
[the remark this answers is not in the document]
Comment (dnadales) [reply cardano-node/test/cardano-node-test.hs:32]: It seems so. Good catch.

File: cardano-node/src/Cardano/Node/Tracing/Tracers/Startup.hs [hunk 2]
```diff
@@ -130,23 +130,23 @@ getStartupInfo nc (SomeConsensusProtocol whichP pForInfo) fp = do
 --
 data ConsensusNetworkVersionTuple a b = ConsensusNetworkVersionTuple a b
 
--- TODO: provide JSON instance for `BlockNodeToClientVersion`
-instance Show blkVersion => ToJSON (ConsensusNetworkVersionTuple NodeToClientVersion blkVersion) where
+instance ToJSON blkVersion => ToJSON (ConsensusNetworkVersionTuple NodeToClientVersion blkVersion) where
     toJSON (ConsensusNetworkVersionTuple nodeToClientVersion blockVersion) =
       Aeson.object [ "nodeToClientVersion" .= toJSON nodeToClientVersion
-                   , "blockVersion" .= String (pack . show $ blockVersion)
+                   , "blockVersion" .= toJSON blockVersion
```
Comment (newhoggy) [inline cardano-node/src/Cardano/Node/Tracing/Tracers/Startup.hs:136]: I think `(.=)` already calls `toJSON` for you.

File: cardano-node/test/Test/Cardano/Tracing/OrphanInstances/HardFork.hs [hunk 3]
```diff
@@ -0,0 +1,155 @@
+-- | Test the JSON encoding of instances in Cardano.Tracing.OrphanInstances.HardFork
+--
+-- We tests the JSON instances by means of Golden tests using [tasty-golden](https://hackage.haskell.org/package/tasty-golden).
```
Comment (newhoggy) [inline cardano-node/test/Test/Cardano/Tracing/OrphanInstances/HardFork.hs:3]: `s/We tests/We test/`

File: cardano-node/test/Test/Cardano/Tracing/OrphanInstances/HardFork.hs [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+
+--------------------------------------------------------------------------------
+-- Helper functions
+--------------------------------------------------------------------------------
+
+goldenTestJSON :: Aeson.ToJSON a => a -> FilePath -> TestTree
+goldenTestJSON valueToEncode goldenFileBaseName =
+    goldenTest
+      goldenFileBaseName
+      -- We expect the contents to the golde file to be small, hence we read them all in one go.
```
Comment (newhoggy) [inline cardano-node/test/Test/Cardano/Tracing/OrphanInstances/HardFork.hs:142]: Typo: `s/golde/golden`

File: cardano-node/test/Test/Cardano/Tracing/OrphanInstances/HardFork.hs [hunk 5]
```diff
@@ -0,0 +1,155 @@
...
+                        , "ntc_HFV1_allDisabled.json"
+                        )
+                      , (  ntc_HFV1_ByronV1
+                        , "ntc_HFV1_ByronV1.json"
+                        )
+                      , (  ntc_HFV1_ByronV1_ShelleyV1
+                        , "ntc_HFV1_ByronV1_ShelleyV1.json")
+                      , (  ntc_HFV1_ByronV1_ShelleyV6_ConwayV2
+                        , "ntc_HFV1_ByronV1_ShelleyV6_ConwayV2.json")
+                      ]
```
[the remark this answers is not in the document]
Comment (newhoggy) [reply cardano-node/test/Test/Cardano/Tracing/OrphanInstances/HardFork.hs:57]: There's more information about tests here: https://github.com/input-output-hk/cardano-node-wiki/wiki/Running-tests

File: cardano-node/test/cardano-node-test.hs [hunk 6]
```diff
@@ -12,20 +12,20 @@ import qualified Test.Cardano.Node.FilePermissions
 #endif
 import qualified Test.Cardano.Node.Json
 import qualified Test.Cardano.Node.POM
+import qualified Test.Cardano.Tracing.OrphanInstances.HardFork
 
 main :: IO ()
 main = do
-  hSetBuffering stdout LineBuffering
-  hSetEncoding stdout utf8
-  defaultMain
```
[the remark this answers is not in the document]
Comment (dnadales) [reply cardano-node/test/cardano-node-test.hs:20]: Reverted, but note that the `where` clause will be at the same indentation level :)

File: cardano-node/test/Test/Cardano/Tracing/OrphanInstances/HardFork.hs [hunk 7]
```diff
@@ -0,0 +1,145 @@
+-- | Test the JSON encoding of instances in Cardano.Tracing.OrphanInstances.HardFork
+--
+-- We test the JSON instances by means of Golden tests using [tasty-golden](https://hackage.haskell.org/package/tasty-golden).
```
Comment (newhoggy) [inline cardano-node/test/Test/Cardano/Tracing/OrphanInstances/HardFork.hs:3]: This comment still says `tasty-golden`.

**Review by newhoggy — APPROVED** [review]

**Pull request merged**